diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt

```diff
@@ -313,6 +314,14 @@ class NewlinesRule(private val configRules: List<RulesConfig>) : Rule("newlines"
     private fun ASTNode.getParentExpressions() =
             parents().takeWhile { it.elementType in chainExpressionTypes && it.elementType != LAMBDA_ARGUMENT }
 
+    private fun ASTNode.containsManyFunctions() =
+            parents().takeWhile { it.elementType == DOT_QUALIFIED_EXPRESSION }
+                    .lastOrNull()
+                    ?.let {
+                        (Regex("""\((.*?)\)""").findAll(it.text).count()) +
```

Comment:
Why are you constantly trying to parse code using regex?) You have the entire AST which is already parsed and structured for you.